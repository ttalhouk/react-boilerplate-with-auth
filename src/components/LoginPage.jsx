import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';



export class LoginPage extends React.Component {
  render () {
    return(
      <div className='box-layout'>
        <div className='box-layout__box'>
          <h1 className='box-layout__title'>Boilerplate Application</h1>
          <p>Add subtitle</p>
          <button
            className='button'
            onClick={this.props.startLogin}>
            Login with Google
          </button>
        </div>
      </div>
    );
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    startLogin: () => dispatch(startLogin())
  }
}

export default connect(null, mapDispatchToProps)(LoginPage);
