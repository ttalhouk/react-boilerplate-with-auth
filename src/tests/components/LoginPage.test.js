import React from 'react';
import {shallow} from 'enzyme';

import {LoginPage} from '../../components/LoginPage.jsx';

describe('<LoginPage />', () => {

  it('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage startLogin={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should call startLogin on button click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
  });
})
