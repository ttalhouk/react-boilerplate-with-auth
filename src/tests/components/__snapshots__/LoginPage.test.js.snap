// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate Application
    </h1>
    <p>
      Add subtitle
    </p>
    <button
      className="button"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
